hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m

```diff
@@ -0,0 +1,332 @@
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFContentSoqlSyncTarget.h"
+#import <SmartSync/SFSmartSyncSyncManager.h>
+#import <SmartSync/SFSmartSyncConstants.h>
+
+// SOAP request
+#define REQUEST_TEMPLATE @"<?xml version=\"1.0\"?>"\
+"<se:Envelope xmlns:se=\"http://schemas.xmlsoap.org/soap/envelope/\">"\
+"<se:Header xmlns:sfns=\"urn:partner.soap.sforce.com\">"\
+"<sfns:SessionHeader><sessionId>%@</sessionId></sfns:SessionHeader>"\
+"</se:Header>"\
+"<se:Body>%@</se:Body>"\
+"</se:Envelope>"
+#define QUERY_TEMPLATE @"<query xmlns=\"urn:partner.soap.sforce.com\" xmlns:ns1=\"sobject.partner.soap.sforce.com\">"\
+"<queryString>%@</queryString></query>"
+#define QUERY_MORE_TEMPLATE @"<queryMore xmlns=\"urn:partner.soap.sforce.com\" xmlns:ns1=\"sobject.partner.soap.sforce.com\">"\
+"<queryLocator>%@</queryLocator>"\
+"</queryMore>"
+#define SOAP_ENDPOINT @"/services/Soap/u/32.0"
+#define SOAP_PATH @""
+
+// Soal request mime type
+#define XML_MIME_TYPE @"text/xml"
+
+// Soap custom header
+#define SOAP_ACTION @"SOAPAction"
+#define SOAP_ACTION_VALUE @"\"\""
+
+// SOAP response
+#define RESULT @"result"
+#define RECORDS @"records"
+#define SF @"sf:"
+#define QUERY_LOCATOR @"queryLocator"
+#define SIZE @"size"
+#define DONE @"done"
+
+// Params for SFSoapSoqlRequest
+#define SESSION_ID @"sessionId"
+#define QUERY @"query"
+#define IS_LOCATOR @"query"
+
+typedef void (^SFSoapSoqlResponseParseComplete) ();
+
+@interface SFRestAPI ()
+
+- (void)removeActiveRequestObject:(SFRestRequest *)request;
```

Comment:
Should we not expose this method?  Or do you think it's a low enough percentage use case to leave it as is?